src/NHibernate/Proxy/NHibernateProxyBuilder.cs

```diff
@@ -393,9 +395,76 @@ private static void ImplementGetLazyInitializer(TypeBuilder typeBuilder, MethodI
 			IL.Emit(OpCodes.Ldnull);
 			IL.Emit(OpCodes.Bne_Un, skipBaseCall);
 
-			IL.Emit(OpCodes.Ldarg_0);
-			EmitCallMethod(IL, OpCodes.Call, method);
-			IL.Emit(OpCodes.Ret);
+			if (method.DeclaringType.IsInterface &&
+			    method.DeclaringType.IsAssignableFrom(parentType))
+			{
+				var interfaceMap = parentType.GetInterfaceMap(method.DeclaringType);
+				var methodIndex = Array.IndexOf(interfaceMap.InterfaceMethods, method);
+				method = interfaceMap.TargetMethods[methodIndex];
+			}
+
+			if (method.IsAbstract)
+			{
+				/*
+				 * return default(<ReturnType>);
+				 */
+				
+				if (!method.ReturnType.IsValueType)
+				{
+					IL.Emit(OpCodes.Ldnull);
+				}
+				else if (method.ReturnType != typeof(void))
+				{
+					var local = IL.DeclareLocal(method.ReturnType);
+					IL.Emit(OpCodes.Ldloca, local);
+					IL.Emit(OpCodes.Initobj, method.ReturnType);
+					IL.Emit(OpCodes.Ldloc, local);
+				}
+
+				IL.Emit(OpCodes.Ret);
+			}
+			else if (method.IsPrivate)
```

Comment:
I was not considering to do the effort of calling a base explicit implementation, since that is already not easily doable in C# code, and previous releases were not supporting it anyway. So no-op would have suited me. But now that it is done and tested, why not keeping it. It is still better.